### DISCUSSION TITLE
adding pictures for instructions

### DISCUSSION THREAD

**Discussion by wmcaffeine** [tatool/tatool-web#157] [Q&A] [answered]
Hi, I'm having issues adding pictures to my modules. Some of them work, but some of them do not. I've published the pictures online and linked them using a direct link, but when I test the module, I get an error message. However, in some of the modules,  I've used pictures without an issue. Is there a limit to the amount of pictures available to be used in the program?

**Comment by cvonbastian** [comment, marked as answer]
Hi wmcaffeine! No, there's no limit to the number of pictures you can use. Could you please copy & paste the specific error message you're getting? Do you host the pictures on your own webserver or using a service like Cloudinary?

**Reply by wmcaffeine** [reply]
Thanks for your answer. I am using Cloudinary to host my images, and the pictures work when I test them on a separate module, but not in my longer one. The error message I am receiving is:

"Tatool encountered an error during the module execution. 

The Instruction Executable currently doesn't support External HTML resources

https://res.cloudinary.com/caffeinewm/image/upload/v1633579606/instructions/2nd-end_lgccub.png"

I have not changed anything from how it worked before, but it somehow is not working anymore. Again, thanks for your time.

**Reply by alocher** [reply]
The error message indicates that Tatool thinks it is looking for HTML pages. Can you verify that you used the correct property "images" and not "pages" as described in the documentation for the Instruction Executable ([here](https://www.tatool-web.com/#!/doc/lib-acc-instruction.html)).

**Reply by wmcaffeine** [reply]
Ah, thanks. That solved the problem. I had all of them as images except that one.
